src/main/io/vtx_tramp.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Created by jflyper */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <ctype.h>
+
+#include "platform.h"
+
+#ifdef VTX_TRAMP
+
+#include "build/debug.h"
+
+#include "common/utils.h"
+
+#include "io/serial.h"
+#include "drivers/serial.h"
+#include "drivers/vtx_var.h"
+#include "io/vtx_tramp.h"
+
+static serialPort_t *trampSerialPort = NULL;
+
+static uint8_t trampCmdBuffer[16];
+
+void trampClearBuffer(uint8_t *buf)
+{
+    for (int i = 0 ; i < 16 ; i++) {
+        buf[i] = 0;
+    }
+}
+
+void trampPrepareBuffer(uint8_t *buf, uint8_t cmd)
+{
+    trampClearBuffer(buf);
+
+    trampCmdBuffer[0] = 15;
+    trampCmdBuffer[1] = cmd;
+}
+
+void trampWriteBuf(uint8_t *buf)
+{
+    serialWriteBuf(trampSerialPort, buf, 16);
+}
+
+void trampChecksum(uint8_t *buf)
+{
+    uint8_t cksum = 0;
+
+    for (int i = 0 ; i < 14 ; i++)
+        cksum += buf[i];
+
+    buf[14] = cksum;
+}
+
+void trampSetFreq(uint16_t freq)
+{
+    if (!trampSerialPort)
+        return;
+
+    trampPrepareBuffer(trampCmdBuffer, 'F');
+    trampCmdBuffer[2] = freq & 0xff;
+    trampCmdBuffer[3] = (freq >> 8) & 0xff;
+    trampChecksum(trampCmdBuffer);
+    trampWriteBuf(trampCmdBuffer);
+}
+
+void trampSetBandChan(uint8_t band, uint8_t chan)
+{
+    if (!trampSerialPort)
+        return;
+
+    trampSetFreq(vtx58FreqTable[band - 1][chan - 1]);
+}
+
+void trampSetRFPower(uint16_t level)
+{
+    if (!trampSerialPort)
+        return;
+
+    trampPrepareBuffer(trampCmdBuffer, 'P');
+    trampCmdBuffer[2] = level & 0xff;
+    trampCmdBuffer[3] = (level >> 8) & 0xff;
+    trampChecksum(trampCmdBuffer);
+    trampWriteBuf(trampCmdBuffer);
+}
+
+void trampSetPitmode(uint8_t onoff)
```

Comment:
trampSetFreq, trampSetRFPower, trampSetPitmode can be implemented by trampSetU16(uint8_t cmd, uint16_t val);